src/clinical/clinical-service.ts

```diff
@@ -313,6 +342,8 @@ export const getClinicalEntityMigrationErrors = async (
   return { migration, clinicalMigrationErrors };
 };
 
+type DonorWithId = Donor & { donorId: number };
```

Comment:
`Donor` has a `donorId` property already
```

export interface Donor {
  _id?: string;
  __v?: number; // mongodb property not being filtered out
  createBy?: string;
  schemaMetadata: SchemaMetadata;
  donorId?: number;
  gender: string;
  submitterId: string;
  programId: string;
  specimens: Array<Specimen>;
  clinicalInfo?: ClinicalInfo;
  primaryDiagnoses?: Array<PrimaryDiagnosis>;
  familyHistory?: Array<FamilyHistory>;
  comorbidity?: Array<Comorbidity>;
  followUps?: Array<FollowUp>;
  treatments?: Array<Treatment>;
  exposure?: Array<Exposure>;
  biomarker?: Array<Biomarker>;
  createdAt?: string;
  updatedAt?: string;
  completionStats?: CompletionStats;
}
``` from src/clinical/clinical-entities.ts

Reply:
But it's `donorId?` which is what's causing the issue

Reply:
I see. the filter for checking number type threw me.
either it's an optional type and you check its not null in code. or it's a mandatory value.


we can do those checks rather than change the type. so you are right with a filter, it can just be a filter checking for existence of value. or setup default values, as I've seen in other code. 

just observations, totally your call on which way you want it to work